Automattic/wp-calypso
# Checkout: use useShoppingCart in contact validation

#### Changes proposed in this Pull Request

The `WPCOMCartItem` type was created as a sort of hack to allow passing `ResponseCartItem` object data through checkout via the `useLineItems` hook that's part of the `composite-checkout` package, since the generic product data available in that hook's types is not sufficient. However, now that `useShoppingCart` is available, we can get the `ResponseCartItem` products directly where they are needed. This will make it easier to manipulate and query cart items in the code since there will only be one data type we need to understand, and since `ResponseCartItem` predates `WPCOMCartItem`, there's already considerable support for it in calypso.

This PR removes `useLineItems` from the contact validation code in checkout, replacing it with `useShoppingCart`.

#### Testing instructions

- Enter checkout with a domain product in your cart.
- Enter invalid contact information and click "Continue" to move to the last step.
- Verify that you get a validation error.
- Correct your contact information and click "Continue" again.
- Verify that there are no errors.

We should then repeat the above when the cart contains only G Suite, but this is harder to test.
